### PR TITLE
sd-device-monitor: introduce two new filters and use them in dissect-image.c

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/systemd/sd-device.h [hunk 1]
```diff
@@ -136,6 +136,7 @@ int sd_device_monitor_stop(sd_device_monitor *m);
 
 int sd_device_monitor_filter_add_match_subsystem_devtype(sd_device_monitor *m, const char *subsystem, const char *devtype);
 int sd_device_monitor_filter_add_match_tag(sd_device_monitor *m, const char *tag);
+int sd_device_monitor_filter_add_match_sysattr(sd_device_monitor *m, const char *sysattr, const char *value, int match);
```
Comment (bluca) [inline src/systemd/sd-device.h:139]: any chance we can add a unit test for this in src/libsystemd/sd-device/test-sd-device-monitor.c ?

Comment (bluca) [reply src/systemd/sd-device.h:139]: same for sd_device_monitor_filter_add_match_parent

File: src/shared/dissect-image.c [hunk 2]
```diff
@@ -312,6 +317,14 @@ static int wait_for_partition_device(
         if (r < 0)
                 return r;
 
+        r = sd_device_monitor_filter_add_match_parent(monitor, parent, true);
```
Comment (bluca) [inline src/shared/dissect-image.c:324]: aren't non-BPF filters problematic from the performance point of view?

File: src/systemd/sd-device.h (hunk 1 above)
Comment (yuwata) [reply src/systemd/sd-device.h:139]: Done.

File: src/shared/dissect-image.c (hunk 2 above)
Comment (yuwata) [reply src/shared/dissect-image.c:324]: In this case, it should not. As the received devices are already filtered by their subsystem and devtype.

**Comment by yuwata** [comment]
@bluca Thank you for comments. Now added tests for filters of sd-device-monitor. PTAL.

**Review by bluca — APPROVED** [review]
lgtm - but we probably want to wait after v248 given we are already in the RC phase?

**Comment by yuwata** [comment]
> lgtm - but we probably want to wait after v248 given we are already in the RC phase?

I do not have any strong opinion to include this v248. So, if no one strongly want this to be included into v248, let's postpone.

**Comment by poettering** [comment]
hmm, i wonder if we really should expose the ability to negate the two matches. unless there's really good immediate usecase I'd drop that for now. We can always add that later on should the need arise. I think the positive matches have a very good usecase: the dissect logic, but the negative matches?

looks great otherwise.

**Comment by yuwata** [comment]
You mean, `sd_device_monitor_filter_add_nomatch_xxx()` can be added later if there is a good usecase?

**Comment by poettering** [comment]
> You mean, `sd_device_monitor_filter_add_nomatch_xxx()` can be added later if there is a good usecase?

precisely. i'd keep things minimal: only add the stuff we know is useful. We can always extend things later.

**Comment by yuwata** [comment]
I see. OK.

File: src/libsystemd/libsystemd.sym [hunk 3]
```diff
@@ -750,4 +750,5 @@ global:
         sd_device_get_seqnum;
         sd_device_new_from_stat_rdev;
         sd_device_trigger;
+        sd_device_monitor_filter_add_match_sysattr;
```
Comment (keszybz) [inline src/libsystemd/libsystemd.sym]: This needs to be moved to a new subsection now.

**Review by keszybz — CHANGES_REQUESTED** [review]

File: src/libsystemd/libsystemd.sym (hunk 3 above)
Comment (yuwata) [reply src/libsystemd/libsystemd.sym]: Done.

**Comment by yuwata** [comment]
Rebased. Upgrading the green label.

**Comment by yuwata** [comment]
Rebased again.

**Pull request merged**